src/main/webapp/app/lecture/pdf-preview/pdf-preview.component.ts

```
@@ -334,8 +400,128 @@ export class PdfPreviewComponent implements OnInit, OnDestroy {
+    /**
+     * Updates the IDs of remaining pages after some have been removed.
+     */
+    updatePageIDs() {
+        const remainingPages = this.pdfContainer.nativeElement.querySelectorAll('.pdf-canvas-container');
+        remainingPages.forEach((container, index) => {
+            const pageIndex = index + 1;
+            container.id = `pdf-page-${pageIndex}`;
+            const canvas = container.querySelector('canvas');
+            const overlay = container.querySelector('div');
+            const checkbox = container.querySelector('input[type="checkbox"]');
+            canvas!.id = String(pageIndex);
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)